### PR TITLE
feat: add e2e-aws for nvidia extensions

### PR DESCRIPTION
**Pull request by frezbo**
Add e2e tests for nvidia

### CODE REVIEW

File: internal/integration/api/extensions_nvidia.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+	}
+
+	nodes := suite.getNVIDIANodes("node.kubernetes.io/instance-type=g4dn.xlarge")
+	for _, node := range nodes {
+		suite.AssertServicesRunning(suite.ctx, node, map[string]string{
+			"ext-nvidia-persistenced": "Running",
+		})
+	}
+
+	// nodes = suite.getNVIDIANodes("node.kubernetes.io/instance-type=p4d.24xlarge")
```
Comment (smira) [inline internal/integration/api/extensions_nvidia.go:80]: is it commented out for some reason?

File: internal/integration/base/api.go [hunk 2]
```diff
@@ -480,6 +482,63 @@ func (apiSuite *APISuite) UserDisks(ctx context.Context, node string, sizeGreate
...
+	}
+}
+
+// AssertExpectedModules verifies that expected kernel modules are loaded on the node.
+func (apiSuite *APISuite) AssertExpectedModules(ctx context.Context, node string, expectedModules map[string]string) {
+	nodeCtx := client.WithNode(ctx, node)
+
+	fileReader, err := apiSuite.Client.Read(nodeCtx, "/proc/modules")
+	defer func() {
+		err = fileReader.Close()
```
Comment (smira) [inline internal/integration/base/api.go:506]:
```suggestion
		apiSuite.Require().NoError(fileReader.Close())
```

and it should be below line 509, as we shouldn't close if the `Read` call fails

File: internal/integration/base/api.go [hunk 3]
```diff
@@ -480,6 +482,63 @@ func (apiSuite *APISuite) UserDisks(ctx context.Context, node string, sizeGreate
...
+	var loadedModules []string
+
+	for scanner.Scan() {
+		loadedModules = append(loadedModules, strings.Split(scanner.Text(), " ")[0])
+	}
+	apiSuite.Require().NoError(scanner.Err())
+
+	fileReader, err = apiSuite.Client.Read(nodeCtx, fmt.Sprintf("/lib/modules/%s/modules.dep", constants.DefaultKernelVersion))
+	defer func() {
+		err = fileReader.Close()
```
Comment (smira) [inline internal/integration/base/api.go:522]: same as above

File: internal/pkg/extensions/kernel_modules.go [hunk 4]
```diff
@@ -132,7 +132,7 @@ func GenerateKernelModuleDependencyTreeExtension(extensionsPathWithKernelModules
 		return mount.Unmount(overlays)
 	})
 
-	log.Println("running depmod to generate kernel modules dependency tree")
+	printFunc("running depmod to generate kernel modules dependency tree")
```
Comment (smira) [inline internal/pkg/extensions/kernel_modules.go:135]: I was going to fix this, thanks for fixing it!

File: internal/integration/api/extensions_nvidia.go (hunk 1 above)
Comment (frezbo) [reply internal/integration/api/extensions_nvidia.go:80]: yes, the a100 gpu's are so costly compared to aws, so want to do that only in azure 30$/hr vs 3$/hr

**Review by smira — APPROVED** [review]

**Pull request merged**